uniconnect-epfl/uniconnect
# fix: upgraded expo sdk and fix some small issues

# What I did
- Upgraded expo SDK to 51
- Upgraded dependencies that the new SDK needed.
- Fixed issues with the layout of search bars.
- Fixed layout issues in the profile screen.
- Fixed issues with camera authorizations on some devices.
- Fixed a problem with the QR scanning debouncer.
- Fixed a problem with the navigation when scanning a QR code.

# How I did it
- Updated expo SDK and dependencies.
- I improved the `InputField` component and used it for all search bars so the layout is uniform and function everywhere in the app.
- I changed some of the registration layout because some of it got broken when I modified the InputField.
- Changed the styles in `MyProfile` and  `globalProfileStyles` to fix all layout issues in the `MyProfile`  screen.
- Added the camera authorizations as plugin.
- I also removed some unused styles.
- Modified a wrong navigation parameter after having scanned a QR code.

# How to verify it
Open the app in expo go and:
- Check that the search bars function correctly in `HomeScreen`, `ContactList`, `ContactGraph` and `ExternalProfileScreen`
- Go to `MyProfile` and check that the layout is good.
- Try to use the camera to scan someone's QR code (note that if the QR does not come from the same Expo go session, it will say "Qr not recognized" but if the camera scans successfully, it means it is working). **I need someone to test this for me becuase it always worked on my device.**

# Demo video

https://github.com/uniconnect-epfl/uniconnect/assets/93329823/9836e978-1c2e-4d82-8294-ea911915e14b

# Pre-merge checklist
The changes I have introduced:
